### PR TITLE
Normalize share name to not include capital letters

### LINKED ISSUE #100729: Flaking pull-kubernetes-e2e-aks-engine-azure-file Invalid Resource Name
#### Which jobs are flaking:

pull-kubernetes-e2e-aks-engine-azure-file 
pull-kubernetes-e2e-azure-file-windows

#### Which test(s) are flaking:

AzureFile CSI Driver End-to-End Tests: Dynamic Provisioning should create a volume on demand with mount options [kubernetes.io/azure-file] [file.csi.azure.com] [Windows] 
AzureFile CSI Driver End-to-End Tests: Dynamic Provisioning should create multiple PV objects, bind to PVCs and attach all to different pods on the same node [kubernetes.io/azure-file] [file.csi.azure.com] [Windows]
AzureFile CSI Driver End-to-End Tests: Dynamic Provisioning should create a volume on demand and mount it as readOnly in a pod [kubernetes.io/azure-file] [file.csi.azure.com] [Windows] 

#### Testgrid link:
https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-aks-engine-azure-file

#### Reason for failure:
If the randomly generated kubetest resource group has capital letters, the share name will contain invalid characters and fail provisioning. 

#### Anything else we need to know:
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

### PR DESCRIPTION
**Pull request by kassarl**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
Normalizes the generated share name to contain only lower case letters, as per the [Azurefile share name rules](https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-shares--directories--files--and-metadata)

#### Which issue(s) this PR fixes:

Fixes #100729 

#### Does this PR introduce a user-facing change?

```release-note
azurefile: Normalize share name to not include capital letters
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/volume/azure_file/azure_provision.go [hunk 1]
```diff
@@ -198,10 +198,10 @@ func (a *azureFileProvisioner) Provision(selectedNode *v1.Node, allowedTopologie
 	}
 
 	if shareName == "" {
-		// File share name has a length limit of 63, and it cannot contain two consecutive '-'s.
+		// File share name has a length limit of 63, it cannot contain two consecutive '-'s, and all letters must be lower case.
 		name := util.GenerateVolumeName(a.options.ClusterName, a.options.PVName, 63)
-		shareName = strings.Replace(name, "--", "-", -1)
-	}
+		shareName = strings.ToLower(strings.Replace(name, "--", "-", -1))
+}
```
Comment (wzshiming) [inline pkg/volume/azure_file/azure_provision.go:204]:
```suggestion
		shareName = strings.Replace(name, "--", "-", -1)
		shareName = strings.ToLower(shareName)
	}
```

**Review by andyzhangx — COMMENTED** [review]
/kind bug
/priority important-soon
/sig cloud-provider
/area provider/azure
/triage accepted
/ok-to-test

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
/retest

**Comment by ialidzhikov** [comment]
@andyzhangx , does it make sense to backport this change?

**Comment by andyzhangx** [comment]
> @andyzhangx , does it make sense to backport this change?

@ialidzhikov already in backporting now.

**Pull request merged**